lib/Monads/nondet/Nondet_VCG.thy

```diff
@@ -186,51 +213,65 @@ lemma hoare_conj:
   "\<lbrakk> \<lbrace>P\<rbrace> f \<lbrace>Q\<rbrace>; \<lbrace>P'\<rbrace> f \<lbrace>Q'\<rbrace> \<rbrakk> \<Longrightarrow> \<lbrace>P and P'\<rbrace> f \<lbrace>Q and Q'\<rbrace>"
   unfolding valid_def by auto
 
-lemmas hoare_post_taut = hoare_TrueI (* FIXME lib: eliminate *)
-
-lemmas wp_post_taut = hoare_TrueI[where P=\<top>]
-lemmas wp_post_tautE = hoareE_TrueI[where P=\<top>]
-
 lemma hoare_pre_cont[simp]:
   "\<lbrace>\<bottom>\<rbrace> f \<lbrace>P\<rbrace>"
   by (simp add:valid_def)
 
+lemma hoare_FalseE[simp]:
+  "\<lbrace>\<bottom>\<rbrace> f \<lbrace>Q\<rbrace>, \<lbrace>E\<rbrace>"
+  by (simp add: valid_def validE_def)
+
+\<comment> \<open>FIXME: remove?\<close>
 lemma hoare_return_drop_var[iff]:
   "\<lbrace>Q\<rbrace> return x \<lbrace>\<lambda>r. Q\<rbrace>"
```

Comment:
For the following lemmas as well -- if they are unused, then yes we should definitely remove them. Of course the `iff` and `intro` ones might be used implicitly somewhere.